packages/cli/src/services/McpPromptLoader.ts

```diff
@@ -132,9 +133,31 @@ export class McpPromptLoader implements ICommandLoader {
                 };
               }
 
+              const allMessageContents = result.messages.map((m) => m.content);
+
+              const isPureStringPrompt = allMessageContents.every(
+                (content) => typeof content === 'string',
+              );
+
+              if (isPureStringPrompt) {
+                return {
+                  type: 'submit_prompt',
+                  content: (allMessageContents as string[]).join('\n\n'),
+                };
+              }
+
+              const content: PartListUnion = allMessageContents.flatMap(
+                (content) => {
+                  if (typeof content === 'string') {
+                    return [content];
+                  }
+                  return content as PartListUnion;
```

Comment:
I tried passing through the model roles locally here, by just returning the full `response.messages` (a `PromptMessage[]`) and fwiw this line causes that to fail much later on.

This cast should be removed as it hides bugs and instead we should fix the part in the code that is actually requiring the case (where `PromptMessage[]` is added as a valid type). Or, we should handle that case here (probably ideal).

Reply:
Actually, what I think we want to to change this to a ContentUnion, so we can attach the roles, and then map the PromptMessage types to the Content type?